pkg/workloads/spark_job/spark_util.py

```diff
@@ -520,12 +582,13 @@ def validate_transformer(column_name, df, ctx, spark):
                 )
 
             # perform the necessary upcast/downcast for the column e.g INT -> LONG or DOUBLE -> FLOAT
-            transform_spark_df = transform_spark_df.withColumn(
-                column_name,
-                F.col(column_name).cast(
-                    CORTEX_TYPE_TO_SPARK_TYPE[ctx.transformed_columns[column_name]["type"]]
-                ),
-            )
+            if not skip_validation:
+                transform_spark_df = transform_spark_df.withColumn(
+                    column_name,
+                    F.col(column_name).cast(
+                        CORTEX_TYPE_TO_SPARK_TYPE[ctx.transformed_columns[column_name]["type"]]
```

Comment:
`ctx.transformed_columns[column_name]["type"]` should probably not be used, and instead the helper should be used (see comment above)